lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -82,21 +84,37 @@ def get(self, request, *args, **kwargs):
             reset_masquerade_data=True
         )
 
+        user_timezone_locale = user_timezone_locale_prefs(request)
+        user_timezone = user_timezone_locale['user_timezone']
+
         transformers = BlockStructureTransformers()
         transformers += course_blocks_api.get_course_block_access_transformers(request.user)
         transformers += [
             BlocksAPITransformer(None, None, depth=3),
         ]
-        get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
         course_blocks = get_course_blocks(request.user, course_usage_key, transformers, include_completion=True)
 
         enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
 
+        course_grade = CourseGradeFactory().read(request.user, course)
```

Comment:
I do not love that this is named a CourseGradeFactory as that is our convention for testing.